### PR TITLE
new Flatpak for Vim/GVim

### PR DESCRIPTION
**Pull request by ramcq**

### CODE REVIEW

File: org.vim.Vim.json [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+          "--disable-xsmp",
+          "--disable-xsmp-interact",
+          "--with-x"
+      ],
+      "post-install": [
+          "install -D -m644 org.vim.Vim.appdata.xml /app/share/appdata/org.vim.Vim.appdata.xml",
+          "install -D -m644 runtime/vim.desktop /app/share/applications/org.vim.Vim.desktop",
+          "install -D -m644 runtime/gvim.desktop /app/share/applications/org.vim.Vim.GVim.desktop",
+          "install -D -m644 runtime/vim48x48.png /app/share/icons/hicolor/48x48/apps/org.vim.Vim.png",
+          "sed -i s/^Icon=.*/Icon=org.vim.Vim/ /app/share/applications/*.desktop"
```
Comment (TingPing) [inline org.vim.Vim.json]: `desktop-file-edit` is much cleaner =)

File: org.vim.Vim.json [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+  "app-id": "org.vim.Vim",
+  "branch": "stable",
+  "runtime": "org.freedesktop.Platform",
+  "runtime-version": "1.6",
+  "sdk": "org.freedesktop.Sdk",
+  "command": "vim",
+  "cleanup": [
+      "/share/man"
+  ],
+  "finish-args": [
```
Comment (TingPing) [inline org.vim.Vim.json]: I imagine you want filesystem access.

File: org.vim.Vim.appdata.xml [hunk 3]
```diff
@@ -0,0 +1,61 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!-- Copyright 2014 Richard Hughes <richard@hughsie.com> -->
+<!--
+EmailAddress: Bram@moolenaar.net>
+SentUpstream: 2014-05-22
+-->
+<application>
```
Comment (TingPing) [inline org.vim.Vim.appdata.xml]: Adding `<releases>` would be nice.

File: org.vim.Vim.json [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+    "--share=ipc",
+    "--socket=x11",
+    "--socket=wayland"
+  ],
+  "modules": [
+    {
+      "name": "vim",
+      "config-opts": [
+          "--enable-gui=gtk3",
```
Comment (Barthalion) [inline org.vim.Vim.json]: Looking through plugins I use, lua from `shared-modules` would be nice to have.

**Comment by nedrichards** [comment]
bot, build org.vim.Vim

**Comment by nedrichards** [comment]
I see that the appdata has the tag `SentUpstream: 2014-05-22` do you know if there's any progress there? Or if a swift poke is required.

**Comment by ramcq** [comment]
Fixed LUA, icons, filesystem access, <releases> and a few other tidy-ups.

**Comment by TingPing** [comment]
bot, build org.vim.Vim

**Comment by ramcq** [comment]
> The appstream appears to be missing the required <name> element. I'm not quite sure how that got through appstream validate, or do we only run that on real commits?

Boggle. Added.

**Comment by nedrichards** [comment]
bot, build org.vim.Vim

**Comment by nedrichards** [comment]
Repository has been created: https://github.com/flathub/org.vim.Vim

**Pull request closed without merging**